File: Week 02/id_237/groupAnagrams.java
//给定一个字符串数组，将字母异位词组合在一起。字母异位词指字母相同，但排列不同的字符串。

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for(String str : strs){
            char[] temp = str.toCharArray();
            Arrays.sort(temp);
            String keystr = String.valueOf(temp);
            if(! map.containsKey(keystr))
                map.put(keystr,new ArrayList<String>());
            map.get(keystr).add(str);
        }
        return new ArrayList<>(map.values());
    }
}

File: Week 02/id_262/group-anagrams.py
from typing import List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        obj = {}
        for s in strs:
            obj[tuple(sorted(s))] = obj.get(tuple(sorted(s)), []) + [s]
        return list(obj.values())
